File: java_app/sc.java
import java.io.FileWriter;
import java.io.IOException;

public class sc {
    public static void main(String[] args) {
        int N = 100000000; 
        long startTime = System.currentTimeMillis();

        boolean[] isPrime = new boolean[N + 1];
        for (int i = 0; i <= N; i++) {
            isPrime[i] = true;
        }
        isPrime[0] = isPrime[1] = false;

        int sqrtN = (int) Math.sqrt(N);
        for (int i = 2; i <= sqrtN; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= N; j += i) {
                    isPrime[j] = false;
                }
            }
        }

        long endTime = System.currentTimeMillis();
        long executionTime = endTime - startTime;

        try (FileWriter writer = new FileWriter("output.txt")) {
            for (int i = 2; i <= N; i++) {
                if (isPrime[i]) {
                    writer.write(i + "\n");
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
     
        System.out.println(executionTime);
    }
}

File: js_app/sc.js
const fs = require("fs");

function sieveOfEratosthenes(n) {
  const isPrime = new Uint8Array(n + 1);
  isPrime.fill(1);
  isPrime[0] = isPrime[1] = 0;
  const sqrtN = Math.floor(Math.sqrt(n));

  for (let i = 2; i <= sqrtN; i++) {
    if (isPrime[i]) {
      for (let j = i * i; j <= n; j += i) {
        isPrime[j] = 0;
      }
    }
  }
  const primes = [];
  for (let i = 2; i <= n; i++) {
    if (isPrime[i]) primes.push(i);
  }
  return primes;
}

function main() {
  const N = 100000000;
  const startTime = Date.now();
  const primes = sieveOfEratosthenes(N);
  const endTime = Date.now();
  const executionTime = endTime - startTime;
  fs.writeFileSync("output.txt", primes.join("\n"));

  console.log(executionTime);
}

main();
